deploy/terraform-datacommons-website/modules/gke/main.tf

```diff
@@ -13,24 +13,47 @@
 # limitations under the License.
 
 locals {
-    # Example cluster name: datacommons-us-central1
-    cluster_name = format("%s-%s%s",var.cluster_name_prefix,var.region, var.resource_suffix)
+    # Example cluster name: datacommons-us-central1 or datacommons-us-central1-a
+    cluster_name = format("%s-%s%s",var.cluster_name_prefix,var.location, var.resource_suffix)
 }
 
-resource "null_resource" "gke_cluster" {
-  provisioner "local-exec" {
-    command = "sh create_cluster.sh"
-    working_dir = path.module
+resource "google_container_cluster" "primary" {
+  name     = local.cluster_name
+  location = var.location
 
-    environment = {
-      PROJECT_ID   = var.project_id
-      CLUSTER_NAME = local.cluster_name
-      NODES        = var.num_nodes
-      REGION       = var.region
-    }
+  # We can't create a cluster with no node pool defined, but we want to only use
+  # separately managed node pools. So we create the smallest possible default
+  # node pool and immediately delete it.
+  remove_default_node_pool = true
+  initial_node_count       = 1
+  networking_mode          = "VPC_NATIVE"
+
+  workload_identity_config {
+    workload_pool = "${var.project_id}.svc.id.goog"
+  }
+
+  ip_allocation_policy {
+    cluster_ipv4_cidr_block = "/14"
   }
 }
 
+resource "google_container_node_pool" "gke_node_pools" {
+  name       = local.cluster_name
+  location   = var.location
+  cluster    = google_container_cluster.primary.name
+  node_count = var.num_nodes
+
+  node_config {
+    machine_type = "e2-highmem-4"
```

Comment:
can you check if we can use a weaker machine_type?

Reply:
Following our kustomize template, I'm changing the memory from website container 8G -> 3G, ESP 2G -> 1G. Which will be something like 12.75G occupied of 15.X G available in a e2-standard-4. 

However, the cost saving is only $30 or so (from ~$132 to ~$98). If there's anything that we shouldn't cheap out  on imo it's the machine type, because k8s under pressure will give all sorts of weird errors. We may also need more resources for new services. I think keeping this machine type is appropriate but will leave the final desicion to you.

Reply:
Then sounds good to keep this. In that case, can revert the mem change above?

Reply:
Keeping the mem changes for now to conserve resources for other potential pods (like argo). Please see the response to Julia's comment.